coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/OperatorTransformer.java

```diff
@@ -0,0 +1,350 @@
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import static com.linkedin.coral.common.calcite.CalciteUtil.*;
+
+
+/**
+ * Object for transforming Operator from one SQL language to another SQL language at the SqlNode layer.
+ *
+ * Suppose f1(a1, a2, ..., an) in the first language can be computed by
+ * f2(b1, b2, ..., bm) in the second language as follows:
+ *    (b1, b2, ..., bm) = g(a1, a2, ..., an)
+ *    f1(a1, a2, ..., an) = h(f2(g(a1, a2, ..., an)))
+ *
+ * We need to define two transformation functions:
+ * - A vector function g for transforming all operands
+ * - A function h for transforming the result.
+ *
+ * This class will represent g and h as expressions in JSON format as follows:
+ * - Operators: +, -, *, /, and ^
+ * - Operands: source operands and literal values
+ *
+ * There may also be situations where a function in one language can map to more than one functions in the other
+ * language depending on the set of input parameters.
+ * We define a set of matching functions to determine what function name is used.
+ * Currently, there is no use-case more complicated than matching a parameter string to a static regex.
+ *
+ * Example 1:
+ * In Calcite SQL, TRUNCATE(aDouble, numDigitAfterDot) truncates aDouble by removing
+ * any digit from the position numDigitAfterDot after the dot, like truncate(11.45, 0) = 11,
+ * truncate(11.45, 1) = 11.4
+ *
+ * In Trino, TRUNCATE(aDouble) only takes one argument and removes all digits after the dot,
+ * like truncate(11.45) = 11.
+ *
+ * The transformation from Calcite TRUNCATE to Trino TRUNCATE is represented as follows:
+ * 1. Trino name: TRUNCATE
+ *
+ * 2. Operand transformers:
+ * g(b1) = a1 * 10 ^ a2, with JSON format:
+ * [
+ *  { "op":"*",
+ *    "operands":[
+ *      {"input":1}, // input 0 is reserved for result transformer. source inputs start from 1
+ *      { "op":"^",
+ *        "operands":[
+ *          {"value":10},
+ *          {"input":2}]}]}]
+ *
+ * 3. Result transformer:
+ * h(result) = result / 10 ^ a2
+ * { "op":"/",
+ *    "operands":[
+ *      {"input":0}, // input 0 is for result transformer
+ *      { "op":"^",
+ *        "operands":[
+ *          {"value":10},
+ *          {"input":2}]}]}]
+ *
+ *
+ * 4. Operator transformers:
+ * none
+ *
+ * Example 2:
+ * In Calcite, there exists a hive-derived function to decode binary data given a format, DECODE(binary, scheme).
+ * In Trino, there is no generic decoding function that takes a decoding-scheme.
```

Comment:
Still references to `Calcite` and `Trino`.

Reply:
Is this a problem in the examples? Although the it's a generic class to express operator transformers, the examples should specify the language. In this case, the transformer just happens to be from Trino to Calcite.

Reply:
I am referring to the words of `Calcite` and `Trino` which I believe should not occur in this class.

Reply:
In this comment they appeared in the comments (JavaDoc).

Reply:
If I took out `Calcite` and `Trino` from the comments the examples wouldn't make sense. For example, this comment from the examples: 
`The transformation from Calcite TRUNCATE to Trino TRUNCATE is represented as follows`

Do you mean instead I should just generalize it and say something along the lines of `The transformation of TRUNCATE from one SQL language to another is represented as follows`?

Reply:
Yes, that is one step towards where I think this should go. We can improve `SQL language` to `SQL operator in one IR to SQL operators in another IR`. I expect Coral IR will be used both on the right and left hand side of this map (once in language to Coral and once in Coral to language). If so, we cannot even attribute one of the two sides (right or left) to Coral. Do you agree @ljfgem?

Reply:
Agreed.